### PR TITLE
inital move to hvr ppa

### PR DESCRIPTION
**Pull request by ChristopherBiscardi**
This is a holding PR while we move to hvr's ppa. 7.10.1 seems to build
successfully, so if you're itching for a 7.10 image it looks like this
will work.

This also updates 7.8.3 to 7.8.4 with hvr's ppa.

### CODE REVIEW

**Comment by ChristopherBiscardi** [comment]
@psftw did you have some scripts to test hackage package installs?

File: 7.10/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+## custom apt-get install options
+ENV OPTS_APT        -y --no-install-recommends
+
+## ensure locale is set during build
+ENV LANG            C.UTF-8
+
+RUN apt-get update && \
+    apt-get install ${OPTS_APT} software-properties-common && \
+    add-apt-repository -y ppa:hvr/ghc && \
+    sed -i s/jessie/trusty/g /etc/apt/sources.list.d/hvr-ghc-jessie.list
```
Comment (tianon) [inline 7.10/Dockerfile]: Why not skip `software-properties-common` and just do something like:
```Dockerfile
RUN echo 'deb http://ppa.launchpad.net/hvr/ghc/ubuntu trusty main' > /etc/apt/sources.list.d/ghc.list
```

Comment (ChristopherBiscardi) [inline 7.10/Dockerfile]: indeed. I've updated it as such.

**Comment by psftw** [comment]
Dockerfiles LGTM, and are much better than what I was toying around with last week.  I do wonder about using a PPA for Ubuntu Trusty against Debian Jessie, but it seems to be working fine.  Nitpick: maybe we should set the DEBIAN_FRONTEND variable just for the commands which actually use APT vs. persistently via a layer? 

So far the "cabal install" testing for both tags is going smoothly.  Once that finishes (1-3hr?) I'll report back here and give an up or down vote.

**Comment by md5** [comment]
> Nitpick: maybe we should set the DEBIAN_FRONTEND variable just for the commands which actually use APT vs. persistently via a layer?

Definitely :+1: 

Setting this in a layer can have unintended consequences if someone tries to run an `apt-get` command in a shell inside a `haskell` container. I actually don't think `DEBIAN_FRONTEND` needs to be set at all. I'm not aware of any other official images setting it (for the reason I mentioned, I believe).

**Comment by psftw** [comment]
TL;DR I'm :+1: on this change, though I would suggest that we add `libsqlite3-0` and `libsqlite3-dev` packages.

The results of "cabal install" testing came out similar to the last time I ran these tests, which was great.  As expected, 7.10 has a few more failures caused by packages that haven't been updated for the new GHC, and it was easy to find github issues for these.  As usual, things that depend on X/OpenGL/specific-uncommon-lib failed.

**7.8**

    $ docker ps -af exited=1
    CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS                         PORTS               NAMES
    26e724b21ba1        haskell:cabal-7.8   "cabal install OpenG   About an hour ago   Exited (1) About an hour ago                       cabal-7.8-OpenGL           
    27011f8bc13e        haskell:cabal-7.8   "cabal install cairo   About an hour ago   Exited (1) About an hour ago                       cabal-7.8-cairo            
    7e4ff0826e18        haskell:cabal-7.8   "cabal install wai-a   About an hour ago   Exited (1) About an hour ago                       cabal-7.8-wai-app-static   
    99ec5d51cb29        haskell:cabal-7.8   "cabal install git-a   About an hour ago   Exited (1) About an hour ago                       cabal-7.8-git-annex        
    bb44a8f2b316        haskell:cabal-7.8   "cabal install OpenG   About an hour ago   Exited (1) About an hour ago                       cabal-7.8-OpenGLRaw        

**7.10**

    $ docker ps -af exited=1
    CONTAINER ID        IMAGE                COMMAND                CREATED             STATUS                         PORTS               NAMES
    c1c7e6c35b5c        haskell:cabal-7.10   "cabal install persi   21 minutes ago      Exited (1) 17 minutes ago                          cabal-7.10-persistent-sqlite   
    1c38ea0f3ae2        haskell:cabal-7.10   "cabal install postg   24 minutes ago      Exited (1) 21 minutes ago                          cabal-7.10-postgresql-simple   
    5ebe34fdedfa        haskell:cabal-7.10   "cabal install snap"   26 minutes ago      Exited (1) 26 minutes ago                          cabal-7.10-snap                
    8fa1a6c497bc        haskell:cabal-7.10   "cabal install OpenG   40 minutes ago      Exited (1) 39 minutes ago                          cabal-7.10-OpenGL              
    6aa610583cbe        haskell:cabal-7.10   "cabal install cairo   44 minutes ago      Exited (1) 43 minutes ago                          cabal-7.10-cairo               
    ef012405ca09        haskell:cabal-7.10   "cabal install egiso   About an hour ago   Exited (1) About an hour ago                       cabal-7.10-egison              
    e6bbfa646268        haskell:cabal-7.10   "cabal install git-a   About an hour ago   Exited (1) About an hour ago                       cabal-7.10-git-annex           
    555da630e8ab        haskell:cabal-7.10   "cabal install OpenG   About an hour ago   Exited (1) About an hour ago                       cabal-7.10-OpenGLRaw           
    02344b8c5741        haskell:cabal-7.10   "cabal install pando   2 hours ago         Exited (1) 2 hours ago                             cabal-7.10-pandoc              
    875a11420004        haskell:cabal-7.10   "cabal install case-   2 hours ago         Exited (1) 2 hours ago                             cabal-7.10-case-insensitive    
    fbec5ed1c2fc        haskell:cabal-7.10   "cabal install hakyl   2 hours ago         Exited (1) 2 hours ago                             cabal-7.10-hakyll              
    b802b6f8537c        haskell:cabal-7.10   "cabal install ghc-m   3 hours ago         Exited (1) 3 hours ago                             cabal-7.10-ghc-mod             


Currently, we don't bundle sqlite3+headers, which causes persistent-sqlite (pulled in by git-annex and others) to fail.  I think it's worth the ~3MB to add `libsqlite3-0` and `libsqlite3-dev` to the package list.

**Comment by psftw** [comment]
@md5 Yes, I'm for removing the unnecessary DEBIAN_FRONTEND business altogether, too.

**Comment by ChristopherBiscardi** [comment]
@psftw I'm +1 on `libsqlite*`. 3mb is very little for this image. Thanks a bunch for running the tests.

I'll pull the `DEBIAN_FRONTEND` ENV as well.

**Comment by ChristopherBiscardi** [comment]
@md5 @psftw I removed the `DEBIAN_FRONTEND` and added `libsqlite*`

**Comment by md5** [comment]
LGTM. I'd probably remove the `ENV OPTS_APT` line too and just list those options inline, but that's not a big deal.

**Comment by psftw** [comment]
@md5 actually putting together another PR to fix a couple of minor things before pushing for an official image update, so I'll include that.  stay tuned.
